gpjax/pytree.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2022 The JaxLinOp Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import abc
+import jax
+
+from typing import Any
+
+
+class Pytree(metaclass=abc.ABCMeta):
```

Comment:
This is my bad. But I think we should call this `PyTree` not `Pytree`. Additionally, we should open an issue of putting this in a small library of JAX utilities - would make it easier to maintain `JaxGaussianProcesses`.